packages/legend-graph/src/graph-manager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_MappingBuilderHelper.ts

```diff
@@ -212,20 +217,41 @@ export const V1_buildMappingInclude = (
   context: V1_GraphBuilderContext,
   parentMapping: Mapping,
 ): MappingInclude => {
-  const includedMapping = new MappingInclude(
-    parentMapping,
-    context.resolveMapping(mappingInclude.includedMapping),
-  );
-  if (mappingInclude.sourceDatabasePath && mappingInclude.targetDatabasePath) {
-    includedMapping.storeSubstitutions.push(
-      new SubstituteStore(
-        includedMapping,
-        context.resolveStore(mappingInclude.sourceDatabasePath),
-        context.resolveStore(mappingInclude.targetDatabasePath),
-      ),
+  if (mappingInclude instanceof V1_MappingIncludeMapping) {
+    const includedMapping = new MappingIncludeMapping(
+      parentMapping,
+      context.resolveMapping(mappingInclude.includedMapping),
     );
+    if (
+      mappingInclude.sourceDatabasePath &&
+      mappingInclude.targetDatabasePath
+    ) {
+      includedMapping.storeSubstitutions.push(
+        new SubstituteStore(
+          includedMapping,
+          context.resolveStore(mappingInclude.sourceDatabasePath),
+          context.resolveStore(mappingInclude.targetDatabasePath),
+        ),
+      );
+    }
+    return includedMapping;
+  }
+  const extraIncludeMappingBuilders = context.extensions.plugins.flatMap(
+    (plugin) =>
+      (
+        plugin as DSL_Mapping_PureProtocolProcessorPlugin_Extension
+      ).V1_getExtraMappingIncludeBuilders?.() ?? [],
+  );
+  for (const builder of extraIncludeMappingBuilders) {
+    const builtInclude = builder(mappingInclude, parentMapping, context);
+    if (builtInclude) {
+      return builtInclude;
+    }
   }
-  return includedMapping;
+  throw new UnsupportedOperationError(
```

Comment:
TODO: As you introduce this new type of abstract, we should do something similar to https://github.com/finos/legend-studio/pull/2226